### ISSUE TITLE
[listing] Sometimes see a "Please refresh..." error message after creating a listing

### ISSUE THREAD

**Issue by jabby09** [KlimaDAO/klimadao#1891]
A user reported seeing this warning message after creating a listing with this note:

> Refreshing fixed the issue but thought I'd share in case this is unexpected behavior. It occurred immediately after the 2nd listing transaction finalized.

![image](https://github.com/KlimaDAO/klimadao/assets/103535195/c97b0dc9-1806-403a-8fe0-d7ae02ef37b7)

**To replicate (attempt):**

- [ ] create a listing...go through the modal steps
- [ ] click APPROVE on the create listing modal
- [ ] approve transaction completed and success message appears
- [ ] go to profile page
- [ ] error message above in red appears (note only after creating a 2nd listing)
- [ ] refreshing the profile page cleared the error

**Comment by 0xMakka** [comment]
@jabby09 is this still an issue? I created 3 new listings (on testnet) to try reproduce this but was unsuccessful? I always get the "updating your data" message with a spinner and it eventually resolves.

![Image](https://github.com/KlimaDAO/klimadao/assets/92357475/ea79805a-8070-4f64-abb4-f53a0750f756)

**Comment by jabby09** [comment]
> I actually just got this message but it looks to be correctly formatted message i.e. not saying TypeError: ... etc.

thanks @0xMakka so basically this happens when we can't connect and read data off the blockchain correct?

**Comment by 0xMakka** [comment]
Yeah, from reading the code - it looks to poll the blockchain for an activity timestamp every 2 seconds for a maximum of 50 attempts. If no data is returned in this period of time, this message is displayed.

**Comment by jabby09** [comment]
@Atmosfearful @theprofessora not much more I think we can do with this one except perhaps improve the error message for the user?

**Comment by jabby09** [comment]
@theprofessora assigned this one to you. Need your input on a better error message and perhaps we want to style is differently?

**Comment by theprofessora** [comment]
Generally, if a user can solve an error themselves, then we want we want to offer an immediate CTA.

![image](https://github.com/KlimaDAO/klimadao/assets/110631623/8a38be5b-fe69-4a7b-b8ac-5199a79e1d9d)

here, the "Refresh now" link shouuld refresh the page for them and resolve the issue if possible. also, placing the error box up above helps keep the rest of the design nice and clean. Also i've de-emphasized the technical jargon that most users won't understand. https://www.figma.com/file/tnBViWV19Ln9jjFDekksY7/Profile?type=design&node-id=761-29575&mode=design

**Comment by jabby09** [comment]
> Generally, if a user can solve an error themselves, then we want we want to offer an immediate CTA.

@0xMakka new design to implement here for this error message when you get a chance.
Low priority.

**Comment by Atmosfearful** [comment]
@biwano and @0xtapi have also witnessed this bug. None of us have been able to reproduce it.

The error must be here on this line: https://github.com/KlimaDAO/klimadao/blob/ce2c6dc41f612c86a3a454cec4ac6f60b91f8c46/carbonmark/components/pages/Portfolio/index.tsx#L52

The code here is a bit hard to grok but I think the problem is `carbonmarkUser` can be undefined.
* [ ] First try understand why / when it might be undefined
* [ ] Refactor it to handle that condition
Also check here

https://github.com/KlimaDAO/klimadao/blob/ce2c6dc41f612c86a3a454cec4ac6f60b91f8c46/carbonmark/components/pages/Users/SellerConnected/index.tsx#L93
